### DISCUSSION TITLE
Has anybody got sign-in with apple to work successfully?

### DISCUSSION THREAD

**Discussion by rubanraj7** [diadal/universal-social-auth#34] [Q&A]
I have a vue3 app, with a laravel backend and followed the steps. On desktop, the popup shows up and i sign into apple with 2fa, then the popup closes and nothing happens. There is no redirect. Wondering what the issue could be.

this is my options object:

```
 providers: {
        apple: {
            clientId: 'com.******.www.sid',
            scope: 'name email',
            state: 'random_state_string',
            redirectUri: 'https://www.*****.com/auth/apple/callback',
            responseMode: 'form_post',
            usePopup: true,
            optionalUrlParams: ['response_mode', 'use_popup', 'state'],
        }
}
```

**Comment by diadal** [comment]
Apple support requires an update but, am very busy right PR is welcome

**Reply by rubanraj7** [reply]
What do you need to get it working?

**Reply by diadal** [reply]
time but, am not available at the moment

**Reply by rubanraj7** [reply]
Ok. Please let me know when you do update. 

On another note, I've subscribed to your patreon for the work you've already helped me with to say thank you.

**Reply by diadal** [reply]
Thanks

**Comment by l00rik** [comment]


**Reply by rubanraj7** [reply]
I used the [apple sdk](https://developer.apple.com/documentation/sign_in_with_apple) for the front end, and then sent the response to my backend for verification. 
The SDK will handle the popup/sign up, the back end uses socialite & this [apple provider package ](https://socialiteproviders.com/Apple/#installation-basic-usage). 
I can post the solution later.
